tests/components/transmission/test_config_flow.py

```diff
@@ -0,0 +1,195 @@
+"""Tests for Met.no config flow."""
+from datetime import timedelta
+from unittest.mock import patch
+
+import pytest
+from transmissionrpc.error import TransmissionError
+
+from homeassistant import data_entry_flow
+from homeassistant.components.transmission import config_flow
+from homeassistant.components.transmission.const import (
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+
+from tests.common import MockConfigEntry, mock_coro
+
+NAME = "Transmission"
+HOST = "192.168.1.100"
+USERNAME = "username"
+PASSWORD = "password"
+PORT = 9091
+SCAN_INTERVAL = 10
+
+
+@pytest.fixture(name="api")
+def mock_transmission_api():
+    """Mock an api."""
+    with patch("transmissionrpc.Client", return_value=mock_coro(True)):
```

Comment:
The return value shouldn't be a coroutine, but just a regular mock. The transmission client is a class that returns a regular instance.